src/Framework/AspNetCoreAnalyzers/test/WebApplicationBuilder/DisallowConfigureAppConfigureHostBuilderTest.cs

```diff
@@ -19,192 +23,171 @@ public async Task ConfigurationBuilderRunsWithoutDiagnostic()
-        Assert.Equal("Suggest using WebApplicationBuilder.Configuration instead of ConfigureAppConfiguration", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+    builder.WebHost.{|#0:ConfigureAppConfiguration(builder => builder.AddJsonFile(""foo.json"", optional: true))|};
     }
-    [Fact]
-    public async Task ConfigureAppWebHostOnBuilderProducesDiagnosticInMain()
-    {
-        // Arrange
-        var source = TestSource.Read(@"
+}
+";
+        var fixedSource = @"
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 public static class Test
 {
     public static void Main(string[]args) {
     var builder = WebApplication.CreateBuilder(args);
-    var webhost = builder.WebHost;
-    webhost./*MM*/ConfigureAppConfiguration(builder => { });
-}
+    builder.Configuration.AddJsonFile(""foo.json"", optional: true);
+    }
 }
-");
+";
 
-        // Act
-        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+        var expectedDiagnostic = new DiagnosticResult(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder).WithArguments("ConfigureAppConfiguration").WithLocation(0);
 
         // Assert
-        var diagnostic = Assert.Single(diagnostics);
-        Assert.Same(DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder, diagnostic.Descriptor);
-        AnalyzerAssert.DiagnosticLocation(source.DefaultMarkerLocation, diagnostic.Location);
-        Assert.Equal("Suggest using WebApplicationBuilder.Configuration instead of ConfigureAppConfiguration", diagnostic.GetMessage(CultureInfo.InvariantCulture));
+        await VerifyCS.VerifyCodeFixAsync(source, expectedDiagnostic, fixedSource);
     }
+    
     [Fact]
     public async Task TwoInvocationsProduceTwoDiagnostic()
     {
         // Arrange
-        var source = TestSource.Read(@"
+        var source = @"
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 var builder = WebApplication.CreateBuilder(args);
-builder.Host./*MM1*/ConfigureHostConfiguration(builder =>
-{
-    builder.AddJsonFile(""foo.json"", optional: true);
-});
-builder.WebHost./*MM2*/ConfigureAppConfiguration(builder =>
-{
-    builder.AddJsonFile(""foo.json"", optional: true);
-});
-");
-
-        // Act
-        var diagnostics = await Runner.GetDiagnosticsAsync(source.Source);
+builder.Host.{|#0:ConfigureHostConfiguration(builder => builder.AddJsonFile(""foo.json"", optional: true))|};
+builder.WebHost.{|#1:ConfigureAppConfiguration(builder => builder.AddJsonFile(""foo.json"", optional: true))|};
+";
+        var fixedSource = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+var builder = WebApplication.CreateBuilder(args);
+builder.Configuration.AddJsonFile(""foo.json"", optional: true);
+builder.Configuration.AddJsonFile(""foo.json"", optional: true);
```

Comment:
This doesn't seem like the correct behavior here. I think it's unlikely for users to make the same invocations in different ConfigureX calls, but if they do, it should only produce a single invocation.